source/class/qx/ui/form/FileSelectorButton.js

```diff
@@ -0,0 +1,128 @@
+/* ************************************************************************
+
+   qooxdoo
+
+   https://qooxdoo.org
+
+   Copyright:
+     2022 OETIKER+PARTNER AG
+
+   License:
+     MIT: https://opensource.org/licenses/MIT
+     See the LICENSE file in the project's top-level directory for details.
+
+   Authors:
+     * Tobias Oetiker (oetiker)
+
+************************************************************************ */
+
+/**
+ * The FileSelectionButton opens a window and lets the user pick one or several
+ * files from the local filesystem. A handle is returned which allows access
+ * to the content of the file from javascript. The file can now be processed
+ * in javascript, or it can also be uploaded to a server.
+ * 
+ * *Example*
+ *
+ * Post the content of the file to the server.
+ * 
+ * ```javascript
+ * let button = new qx.ui.form.FileSelectorButton("Select File");
+ * button.addListener('changeFileSelection',function(e){
+ * let fileList = e.getData();
+ *     let form = new FormData();
+ *     form.append('file',fileList[0]);
+ *     let req = new qx.io.request.Xhr("upload",'POST').set({
+ *         requestData: form
+ *     });
+ *     req.addListener('success',(e) => {
+ *         let response = req.getResponse();
+ *     });
+ * });
+ *
+ * [MDN Documentation](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/file)
+ */
+
+qx.Class.define("qx.ui.form.FileSelectorButton", {
+  extend : qx.ui.form.Button,
+  statics: {
+    ID_CNTR: 0
```

Comment:
please can we have a better name for this?  I can't tell without reading the code what it's for, especially as it is a public static

Reply:
switching it to a class variable ...